testkit/src/main/java/org/apache/calcite/test/SqlOperatorTest.java

```diff
@@ -5543,6 +5543,65 @@ private static void checkIf(SqlOperatorFixture f) {
         "INTEGER ARRAY NOT NULL");
   }
 
+  /** Tests {@code MAP_FROM_ARRAYS} function from Spark. */
+  @Test void testMapFromArraysFunc() {
+    final SqlOperatorFixture f0 = fixture();
+    f0.setFor(SqlLibraryOperators.MAP_FROM_ARRAYS);
+    f0.checkFails("^map_from_arrays(array[1, 2], array['foo', 'bar'])^",
+        "No match found for function signature MAP_FROM_ARRAYS\\(<INTEGER ARRAY>, "
+            + "<CHAR\\(3\\) ARRAY>\\)", false);
+
+    final SqlOperatorFixture f = f0.withLibrary(SqlLibrary.SPARK);
+    f.checkScalar("map_from_arrays(array[1, 2], array['foo', 'bar'])", "{1=foo, 2=bar}",
```

Comment:
do all elements of the array have to be of the same type ? could you have [1, "a", true]?

Reply:
@tanclary "do all elements of the array have to be of the same type"  yeap, the args should be find leastRestrictive, otherwise it will throw exception like "SqlValidatorException: Parameters must be of the same type"

Reply:
Ok that could be a good test, then.

Reply:
@tanclary thanks for your quick response, thank you so much